### ISSUE TITLE
Allow users add a dynamic module during a build from source

### ISSUE THREAD

**Issue by alessfg** [nginxinc/ansible-role-nginx#640]
### Discussed in https://github.com/nginxinc/ansible-role-nginx/discussions/638

<div type='discussions-op-text'>

<sup>Originally posted by **tibz7** August  3, 2023</sup>
Is there  a way to use this role and install geoip2 for the opensource version? 
Something like  using the source and passing compilation flags? 
Otherwise the only option i thought of was to not use this role and write my own tasks to compile it with the needed module.</div>

**Comment by olwe0002** [comment]
Hi alessfg,

try 

```yaml
nginx_install_from: source
nginx_static_modules: [http_ssl_module,http_geoip2_module]
```

**Comment by alessfg** [comment]
That parameter is used for NGINX modules following the `--with` pattern listed here http://nginx.org/en/docs/configure.html and given the open source `geoip2` module is considered an external module, you would have to use `--add-module=...` or `--add-dynamic-module=...` instead.

**Comment by tibz7** [comment]
Hello, any progress on this?

**Comment by alessfg** [comment]
Not really, no. Building from source is not recommended and as such building from source options are what we consider a "community" developed features. This does not mean that it will not get worked on, but any other active issue and/or PR is affecting any other feature of the role is prioritized.

PRs are always welcome though!

**Comment by oxpa** [comment]
I'm not sure if Ansible (which is a configuration management tool?) is the best solution for a build automation.
Nginx offers pkg-oss "framework": http://hg.nginx.org/pkg-oss . With it you can build nginx and any module with, like, 3 commands:
```
hg clone http://hg.nginx.org/pkg-oss
cd pkg-oss/$os_flavor
make base module-geoip2
```
You can swap `hg` for `git` if you have proper git module installed.
It doesn't get much easier than that.
And if you need a module which is not in the list of provided "examples" - you can use build_module.sh which will write a stub makefile for you.
Is it really worth maintaining 10 different ways of doing the same?..
